src/main/java/org/breedinginsight/brapps/importer/services/BrAPIQueryService.java

```diff
@@ -0,0 +1,240 @@
+                //TODO: Check that all of the pages were returned
+            } else {
+                // Hit the get endpoint until we get a response
+                Integer accruedWait = 0;
+                ApiResponse<Pair<Optional<T>, Optional<BrAPIAcceptedSearchResponse>>> searchGetResponse = null;
+                while (searchGetResponse == null) {
+                    BrAPIAcceptedSearchResponse searchResult = response.getBody().getRight().get();
+
+                    // TODO: Check if we have more to get for pages
+                    searchGetResponse = searchGetMethod.apply(searchResult.getResult().getSearchResultsDbId(), 0, RESULTS_PER_QUERY);
+                    if (searchGetResponse.getBody().getLeft().isPresent()) {
+                        BrAPIResponse listResponse = (BrAPIResponse) searchGetResponse.getBody().getLeft().get();
+                        BrAPIResponseResult responseResult = (BrAPIResponseResult) listResponse.getResult();
+                        listResult = responseResult != null ? responseResult.getData() :
+                                new ArrayList<>();
+                    } else {
+                        // Wait a bit before we call again
+                        Thread.sleep(SEARCH_WAIT_TIME);
+                        accruedWait += SEARCH_WAIT_TIME;
+                        if (accruedWait >= SEARCH_TIMEOUT) {
+                            throw new ApiException("Search response timeout");
+                        }
+                    }
+                }
+            }
+
+            return listResult;
+        } catch (ApiException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new InternalServerException(e.toString(), e);
+        }
+    }
+
+    private <T> List<T> post(List<T> brapiObjects, Function<List<T>, ApiResponse> postMethod) throws ApiException {
+
+        List<T> listResult = new ArrayList<>();
+        try {
+            ApiResponse response = postMethod.apply(brapiObjects);
+            if (response.getBody() == null) throw new ApiException("Response is missing body");
+            BrAPIResponse body = (BrAPIResponse) response.getBody();
+            if (body.getResult() == null) throw new ApiException("Response body is missing result");
+            BrAPIResponseResult result = (BrAPIResponseResult) body.getResult();
+            if (result.getData() == null) throw new ApiException("Response result is missing data");
+            List<T> data = result.getData();
+            if (data.size() != brapiObjects.size()) throw new ApiException("Number of brapi objects returned does not equal number sent");
+            return data;
+        } catch (ApiException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new InternalServerException(e.toString(), e);
+        }
+    }
+
+    public List<BrAPIGermplasm> getGermplasmByName(List<String> germplasmNames, BrAPIProgram brAPIProgram) throws ApiException {
```

Comment:
Maybe break these specific usage methods into different files or DAOs to limit the size of this file.

Reply:
Yep, agreed.

Reply:
pushed